### DISCUSSION TITLE
Suggestions on UI / UX Enhancements

### DISCUSSION THREAD

**Discussion by ananya-jain1** [Sunbird-Saral/Community#62] [Ideas]
**1. Student Number instead of Page number**

Instead of 'Page Number' on the screen as shown below, can 'Record Number' be shown, as the user is looking at a particular record (row) in the entire page (OMR sheet)?

![image](https://user-images.githubusercontent.com/96038973/208437600-c4c7f96d-531e-4dc3-8306-cef6e1d67303.png)

**2. Padding and alignment to be fixed in UI elements**

Alignment of the profile icon 'D' to be fixed. It doesn't seem to be in the middle of the circle, as shown below.

![image](https://user-images.githubusercontent.com/96038973/208437790-a486a68b-38ba-41bc-a5c9-8def32990ab5.png)

Padding of the close button needs to be fixed. It is coinciding with the mobile screen, as shown below.

![image](https://user-images.githubusercontent.com/96038973/208438236-c603bf1f-3c93-4664-a725-601e9517dbc1.png)

**3. Saving scores post reviewing the same**

Can the scanned scores be saved on the review screen itself? On the review screen (as shown below), there can be a 'Save' button next to the close button to save all the reviewed scores. 'Save' button can be there while reviewing scores of individual student (Screen 1). 'Save All' can be there while reviewing scores of all students (Screen 2).

![image](https://user-images.githubusercontent.com/96038973/208440717-6eb9dbb0-2577-4355-9597-5a8d27e9db02.png)

![image](https://user-images.githubusercontent.com/96038973/208440684-469d243f-c512-400b-858d-3cea04ec926f.png)

**4. below image submit button text**

![Screenshot_2022-12-21-10-56-31-09_f9c6dc39fbf96a0022368795536358b1](https://user-images.githubusercontent.com/91952702/208829348-84b0aad4-1088-4d8e-a888-37eebb22f3e9.jpg)
 need to be change in capital letter

**Comment by AnuragSingh55** [comment]
Adding some additional UI fixes, on top of the ones mentioned above

1. The support button does not trigger any action - can this be dropped from prod for now 

![image](https://user-images.githubusercontent.com/106961941/209766709-477b1dda-d126-4543-a3d3-2d3cccaaeb1a.png)

2. Need to include user prompts, such as “Please mark student attendance before scanning

![image](https://user-images.githubusercontent.com/106961941/209766771-7d1002ad-ec77-4212-98b7-60f1a3910e9e.png)

3. In the review section after scanning, the share button sends a JSON file and doesn’t render UI 

![image](https://user-images.githubusercontent.com/106961941/209766823-d865dcd1-3417-4266-b097-50e42322a29a.png)

4. After scanning the sheets, to save all scans, one has to go back and then click on save all 

- As an alternative can either have a save all scans button after scanning is complete
- Or at least let the user know via prompt that they have to go back and click on save all

![image](https://user-images.githubusercontent.com/106961941/209766914-f42f77f4-0cc1-4121-8a4b-15da2a153c87.png)

5. All grey button seem slightly off, can we have a darker teel shade for them

6. Set drop down needs to be capitalised, also can we add None option in set (if in case assessment has only one set)

**Comment by dileep-gadiraju** [comment]
Hi @AnuragSingh55, please provide more details about "The support button does not trigger any action - can this be dropped from prod for now". If it's a very low-end mobile the dialog may take time to open. Instead of taking a call to drop it directly , we can identify the root cause first.

**Reply by AnuragSingh55** [reply]
Hey @dileep-gadiraju, as discussed today. We can only allow for email option in the support button instead of all the comms apps

**Comment by navin9584** [comment]
Hi @AnuragSingh55 please provide more detail about `In the review section after scanning, the share button sends a JSON file and doesn’t render UI` what is exactly requirement

**Reply by AnuragSingh55** [reply]
Hi @navin9584 , sharing the link for the recreation of this bug - https://drive.google.com/file/d/1zbkwDIG_VdM13aj6ymCT2yVv9h6unWpn/view?usp=sharing.

Can we have the review render a PDF/ or CSV of the proper table instead of this JSON

**Reply by dileep-gadiraju** [reply]
@AnuragSingh55 , the purpose of this share marks is only for tech support to use it as raw data for troubleshooting if any failure in saving marks to backend. This is not reporting functionality. So intentionally its kept as JSON instead of PDF/CSV.

**Comment by dileep-gadiraju** [comment]
Sure, @Navin Nahar ***@***.***> , do the changes accordingly.

**Reply by karsood** [reply]
@dileep-gadiraju - Requesting enhancement with one more aspect of UI:

After scanning when the user is reviewing the scanned results, on hitting 'Next', the page transfers from Student 1 to Student 2 but it **does not** scroll to the top automatically. 
Request: On hitting next, the details for student 2 as listed on top to be shown first and then the user can scroll down to see marks captured.

**Comment by dileep-gadiraju** [comment]
sure @karsood , will be taken care as part of https://project-sunbird.atlassian.net/browse/SS-81
